File: 2020/10/1.js
const fs = require('fs')

console.log('Example:', solution(prepareInput(readInput('example_input'))))
console.log('Solution:', solution(prepareInput(readInput('input'))))

function solution(data) {
  const jolts = data.reduce((acc, jolt, idx, all) => {
    const prev = all[idx - 1]
    const diff = jolt - (prev || 0)
    acc[diff] += 1
    return acc
  }, { 1: 0, 3: 1 })
  return jolts['1'] * jolts['3']
}

function readInput(filePath) {
  return fs.readFileSync(filePath).toString('utf8').trim().split('\n')
}

function prepareInput(lines) {
  return lines.map((line) => parseInt(line.trim(), 10)).sort((a, b) => a - b)
}

File: 2020/06/1.erl
-module(aoc2020_6_1).
-export([main/1]).

read_input(FileName) ->
  {ok, Data} = file:read_file(FileName),
  Content = unicode:characters_to_list(Data),
  string:split(Content, ["\n\n"], all).

prepare_input(Lines) ->
  lists:map(fun(L) -> lists:filter(fun(C) -> C /= 10 end, L) end, Lines).

count_distinct(Str) ->
  F = fun(C, Set) -> sets:add_element(C, Set) end,
  sets:size(lists:foldl(F, sets:new(), Str)).

solution(Data) -> 
  lists:foldl(fun(Str, Sum) -> Sum + count_distinct(Str) end, 0, Data).

main(_Args) ->
  io:format("abcd should be 4: ~w\n", [solution(["abcd"])]),
  io:format("a aa should be 2: ~w\n", [solution(["a", "aa"])]),
  io:format("example: ~w\n", [solution(prepare_input(read_input("example_input.txt")))]),
  io:format("example: ~w\n", [solution(prepare_input(read_input("input.txt")))]).
